third-culture-software/bhima
# Create assets inventory management

This PR does several things related to Assets management
- Moves assignment functions to the Assets Management > Assets Registry page
- Other improvements to the Assets Registry page
- Moves all asset (lot) assignment functions to the Assets Management > Assets Registry page
- Adds Asset Scan Registry page (under Assets Management)
  - Adds ability to add and edit asset scans (in action menu)
  - Adds top-level barcode scanning function to add a new asset scan
- Consolidates all asset-related translation tokens to 'assets.json' files
- Added sample inventory (for assets), assets, and asset scans to bhima_test.
- Adds integration tests for asset scans

Closes https://github.com/IMA-WorldHealth/bhima/issues/6350
Closes https://github.com/IMA-WorldHealth/bhima/issues/6465

Work remaining to be done:
 - Add ability to specify a required date for items to be scanned.  I have a plan for this but want to get this PR in the system before tackling it.
 - Add reports for lists of assets remaining to be scanned
 - Add reports for lists of all assets globally as well as in specific final destinations
 - Eventually refine the concept of locations for scans.  Perhaps use GPS coordinates? (Current "Location" is simply a village)

**TESTING**
- Suggest testing first with bhima_test and then with a production dataset
- If you are not using bhima_test:
  - Make sure there are Contacts (entity) in your database.  Use the Contact management page if necessary. Some production databases are not using Contacts, so this step may be necessary for some sites since Asset assignment is to an entity (contact) which can be a person or place, or department, etc.
  - Use the testing directions from https://github.com/IMA-WorldHealth/bhima/pull/6417 to add a new inventory
  - Add a couple of new assets by doing Stock Entry via Integration and be sure to select the asset inventory type created above.
- Go to the  Assets Management > Assets Registry page
  - Test the various action menu items, the grid functionality, etc
  - Assign one of them to an entity (you created above)
  - Pick an asset and use the Barcode function in the drop-down menu to print the barcode (or copy the code)
- Go to the Assets Management > Assets Scan Registry page
  - Use the big [+New Asset Scan] button and paste in the asset barcode (or scan the printed one).  Fill out the data and submit
  - Test the items in the action menu for the asset scan
  - Review all asset-related French translation items
